autoreadme.go

```diff
@@ -207,6 +229,18 @@ func getDoc(dir string) (Doc, error) {
 	}, nil
 }
 
+func renderExample(ex *doc.Example) string {
+	c := &bytes.Buffer{}
+	_ = format.Node(c, token.NewFileSet(), ex.Code)
+	return fmt.Sprintf(
+		"### %s\n\n%s\nCode:\n\n```\n%s\n```\n\nOutput:\n\n```\n%s\n```\n",
+		ex.Name,
```

Comment:
Maybe this should leave the name out, and start with the `ex.Doc` field.

Reply:
Only include "Output:" if Output ≠ "".

Define an Example struct with Name (copied from `doc.Example`) and Doc run through `fmtDoc`† and a field with the formatted code+output.  Leave the rest up to the user's template to organize and format.

† This may need to be changed to specify the header level to use and maybe even be made a template function so the template can say which header level to use. Unclear what the best approach is here.